scripts/dts/extract/globals.py

```diff
@@ -350,6 +395,16 @@ def extract_controller(node_path, prop, prop_values, index,
             continue
 
         cell_parent = phandles[elem[0]]
+
+        if 'gpio-map' in reduced[cell_parent]['props']:
+            # parent is a gpio 'nexus node' (ie has gpio-map)
+            # controller is to be found in the map, using elem[1] as index
+            gpio_controller = nexus_to_controller(cell_parent, elem[1])
+            if gpio_controller is not 'None':
```

Comment:
Should be `is not None`, not `is not 'None'`. Just `if gpio_controller:` should work though.

Something like this:

```python
if not gpio_controller:
    raise Exception("GPIO controller not found")
cell_parent = gpio_controller
```